### ISSUE TITLE
autoConfig and factory is nil on accessing a HTTP private registry

### ISSUE THREAD

**Issue by hashneo** [cloudfoundry-incubator/garden-linux#27]
I'm getting a crash 
```
	/usr/local/go/src/net/http/server.go:1130 +0xbb
github.com/docker/docker/registry.NewSession(0x0, 0x0, 0xc2087be960, 0x1, 0xc2081c0660, 0x0, 0x0)
	/var/vcap/packages/garden-linux/src/github.com/cloudfoundry-incubator/garden-linux/Godeps/_workspace/src/github.com/docker/docker/registry/session.go:58 +0x75a
github.com/cloudfoundry-incubator/garden-linux/old/linux_backend/repository_fetcher.registryProvider.ProvideRegistry(0xa758b0, 0x1b, 0xc208126b40, 0x1, 0x1, 0xc208973bf9, 0xf, 0x0, 0x0, 0x0, ...)
	/var/vcap/packages/garden-linux/src/github.com/cloudfoundry-incubator/garden-linux/old/linux_backend/repository_fetcher/repository_provider.go:49 +0x2ff
github.com/cloudfoundry-incubator/garden-linux/old/linux_backend/repository_fetcher.(*registryProvider).ProvideRegistry(0xc2080d3f50, 0xc208973bf9, 0xf, 0x0, 0x0, 0x0, 0x0)
```

There is no check on authConfig 
dec := utils.NewHTTPAuthDecorator(authConfig.Username, authConfig.Password)

Also factory is not checked so it will also fail.
factory.AddDecorator(dec)

https://github.com/cloudfoundry-incubator/garden-linux/blob/master/Godeps/_workspace/src/github.com/docker/docker/registry/session.go#L58

**Comment by tsaleh** [comment]
Hi @hashneo - can you show me how to reproduce this crash?

**Comment by hashneo** [comment]
I created a private docker registry with a signed ssl (no self signed) and simply did a ‘cf docker-pull' request.

The private docker registry was behind an nginx server which was terminating the SSL -> registry:5000 

Hope this help
Steven

**Comment by onsi** [comment]
Some Lattice users are seeing this issue as well.  We aren't passing in an `authConfig` - we believe that this particular registry configuration shouldn't need an `authConfig` but Docker explodes without one.  @hashneo - can you confirm that you can clone your image without providing any credentials to the registry?

**Comment by hashneo** [comment]
My registry doesn't need credentials. Its a private registry inside our network, signed with a valid ca ssl cert. With the fix I submitted, the registry works fine.

**Comment by goonzoid** [comment]
Fixed by #47 :tada: :balloon:
